### PR TITLE
Improve KSCrash installation error handling and reporting

### PR DESCRIPTION
**Pull request by GLinnik21**
This PR improves the KSCrash installation process by introducing detailed error handling and updating the installation method to return specific error codes. It also updates the `SampleView` to display alerts for installation failures and refactors relevant methods and tests to support these changes.

### CODE REVIEW

File: Samples/Common/Sources/LibraryBridge/RecordingSample.swift [hunk 1]
```diff
@@ -27,9 +27,33 @@
...
-public class RecordingSample {
-    public static func simpleInstall() {
-        let config = KSCrashConfiguration()
-        KSCrash.shared().install(with: config)
+public struct RecordingSample {
+    public enum InstallationError: Error, LocalizedError {
+        case kscrashError(String)
+        case unexpectedError(String)
+
+        public var errorDescription: String? {
```
Comment (bamx23) [inline Samples/Common/Sources/LibraryBridge/RecordingSample.swift:35]: Does this go to `localizedDescription`?

File: Sources/KSCrashInstallations/KSCrashInstallation.m [hunk 2]
```diff
@@ -284,7 +284,7 @@ - (void)installWithConfiguration:(KSCrashConfiguration *)configuration
             }
         };
 
-        [handler installWithConfiguration:configuration];
+        [handler installWithConfiguration:configuration error:NULL];
```
Comment (bamx23) [inline Sources/KSCrashInstallations/KSCrashInstallation.m:287]: Should we update the `KSCrashInstallation` too?

File: Sources/KSCrashRecording/include/KSCrashError.h [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+static NSErrorDomain const KSCrashErrorDomain = @"KSCrashErrorDomain";
+#endif
+
+typedef
+#ifdef __OBJC__
+    NS_ERROR_ENUM(KSCrashErrorDomain, KSCrashInstallErrorCode)
+#else  /* __OBJC__ */
+    enum
+#endif /* __OBJC__ */
+{ KSCrashInstallErrorAlreadyInstalled = 1,
```
Comment (bamx23) [inline Sources/KSCrashRecording/include/KSCrashError.h:48]: You can call it `KSCrashInstallResultCode` and have `KSCrashInstallResultCodeSuccess = 0`.
In this case `NS_ERROR_ENUM` is not fully accurate option but at least C API would not return `0` that is not a part of this enum.

File: Sources/KSCrashRecording/include/KSCrashError.h [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+#ifdef __OBJC__
+#include <Foundation/Foundation.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef __OBJC__
+static NSErrorDomain const KSCrashErrorDomain = @"KSCrashErrorDomain";
```
Comment (bamx23) [inline Sources/KSCrashRecording/include/KSCrashError.h:39]: Is it okay to have it static and not extern (shared across imports)?

Comment (GLinnik21) [reply Sources/KSCrashRecording/include/KSCrashError.h:39]: We use it all over the place, for instance in `KSCrashReportFields.h`

File: Sources/KSCrashRecording/include/KSCrashError.h (hunk 3 above)
Comment (GLinnik21) [reply Sources/KSCrashRecording/include/KSCrashError.h:48]: About `KSCrashInstallResultCode`. In the Swift interface it's called `KSCrashInstallResult`:

```swift
public struct KSCrashInstallResult : CustomNSError, Hashable, Error {
```

Maybe we should just add `KSCrashInstallErrorNone = 0`?

File: Samples/Common/Sources/LibraryBridge/RecordingSample.swift (hunk 1 above)
Comment (GLinnik21) [reply Samples/Common/Sources/LibraryBridge/RecordingSample.swift:35]: This is part of the `LocalizedError` protocol and the system primarily uses it to create `localizedDescription`. Without it, it just prints an error code. With this, I tried not to expose KSCrash structs to the UI layer.

File: Sources/KSCrashRecording/KSCrash.m [hunk 5]
```diff
@@ -436,6 +444,49 @@ - (NSMutableData *)nullTerminated:(NSData *)data
     return mutable;
 }
 
+- (NSError *)errorForInstallErrorCode:(KSCrashInstallErrorCode)errorCode
+{
+    NSString *errorDescription;
+    switch (errorCode) {
+        case KSCrashInstallErrorNone:
+            errorDescription = @"No error";
+            break;
```
Comment (bamx23) [inline Sources/KSCrashRecording/KSCrash.m:453]: Maybe it's better return nil here, wdyt?

**Review by bamx23 — APPROVED** [review]

**Pull request merged**